### DISCUSSION TITLE
vsock doesn't appear to create socket on host?

### DISCUSSION THREAD

**Discussion by alexellis** [cloud-hypervisor/cloud-hypervisor#6831] [Q&A]
Hi,

I'm implementing an agent to measure usage consumption within a VM, it runs within the guest listening on vsock and is accessed via the host.

With Firecracker, this is working as expected. Firecracker opens a UNIX socket at the specified location when the process starts up, the host can read from that location by sending `CONNECT PORT_NUMBER\n`.

> Once the microvm is started, Firecracker will create and start listening on the AF_UNIX socket at uds_path. Incoming connections will get forwarded to the guest microvm, and translated to AF_VSOCK. The destination port is expected to be specified by sending the text command "CONNECT <port_num>\n", immediately after the AF_UNIX connection is established. Connections initiated from within the guest will be forwarded to AF_UNIX sockets expected to be listening at ./v.sock_<port_num>. I.e. a guest connection to port 52 will get forwarded to ./v.sock_52.

https://github.com/firecracker-microvm/firecracker/blob/main/docs/vsock.md

The agent can then dump its state.

With cloud-hypervisor, no socket is created despite the CLI flag being passed as directed in the docs:

```
sudo cloud-hypervisor --api-socket /tmp/ch.sock --vsock cid=6,socket=/tmp/ch.vsock

$ ls /tmp/ch.sock
/tmp/ch.sock

$ ls /tmp/ch.vsock
ls: cannot access '/tmp/ch.vsock': No such file or directory
```

I've also tried booting up a VM with existing code, and the additional argument populated, it made no difference.

VSOCK is enabled on the guest Kernel for Firecracker, and for cloud-hypervisor. The latest version of cloud-hypervisor is being used from the GitHub Releases page (42). The same host is being used to launch either f/c or c/h.

Please could someone advise on whether this feature is supported? The documentation states that it has parity with or pays homage to Firecracker.

**Comment by alexellis** [comment]
I got around this by using the API to add a vsock, which also worked via ch-remote, I couldn't get it to work when using arguments passed to the cloud-hypervisor binary.

**Comment by rbradford** [comment]
> `sudo cloud-hypervisor --api-socket /tmp/ch.sock --vsock cid=6,socket=/tmp/ch.vsock`

Wouldn't work because you didn't create a VM with this command - just started the VMM on the API socket. A VM is only created if `--kernel` is provided - the vsock socket will be created as part of the VM creation step.
